core/src/main/java/io/grpc/internal/DelayedClientTransport.java

```diff
@@ -286,6 +286,9 @@ final void reprocess(@Nullable SubchannelPicker picker) {
 
     for (final PendingStream stream : toProcess) {
       PickResult pickResult = picker.pickSubchannel(stream.args);
+      if (!pickResult.hasResult()) {
```

Comment:
This is inverted; status is always OK when there isn't a result. Add a test to confirm the behavior.